src/components/experiment-tracking/experiment-primary-toolbar/sliding-animation.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { Transition } from 'react-transition-group';
+
+const directions = {
+  leftToRight: {
+    entering: {
+      opacity: 0.5,
+      transform: 'translateX(50%)',
+      visibility: 'visible',
+    },
+    entered: {
+      opacity: 1,
+      transform: 'translateX(50%)',
+      visibility: 'visible',
+    },
+    exiting: {
+      transform: 'translateX(-50%)',
+      opacity: 0.5,
+      visibility: 'hidden',
+    },
+    exited: {
+      transform: 'translateX(-50%)',
+      opacity: 0,
+      visibility: 'hidden',
+    },
+  },
+  rightToLeft: {
+    entering: {
+      opacity: 0.5,
+      transform: 'translateX(-50%)',
+      visibility: 'visible',
+    },
+    entered: {
+      opacity: 1,
+      transform: 'translateX(-50%)',
+      visibility: 'visible',
+    },
+    exiting: {
+      transform: 'translateX(50%)',
+      opacity: 0.5,
+      visibility: 'hidden',
+    },
+    exited: {
+      transform: 'translateX(50%)',
+      opacity: 0,
+      visibility: 'hidden',
+    },
+  },
+};
+
+export const SlideFromLeftToRight = ({ state, duration, children }) => {
+  const defaultStyle = {
+    transition: `transform ${duration}ms ease-in-out`,
+  };
+
+  return (
+    <Transition in={state} timeout={duration}>
+      {(state) => (
+        <div
+          style={{
+            ...defaultStyle,
+            ...directions['leftToRight'][state],
+          }}
+        >
+          {children}
+        </div>
+      )}
+    </Transition>
+  );
+};
+
+export const SlideFromRightToLeft = ({ state, duration, children }) => {
+  const defaultStyle = {
+    transition: `transform ${duration}ms ease-in-out`,
+  };
+
+  return (
+    <Transition in={state} timeout={duration}>
+      {(state) => (
+        <div
+          style={{
+            ...defaultStyle,
+            ...directions['rightToLeft'][state],
+          }}
+        >
+          {children}
+        </div>
+      )}
+    </Transition>
+  );
+};
```

Comment:
This looks great! Just one thing that I've noticed, we can make this part as one reusable component as most of the code is the same. What do you think?

Reply:
hey yeah, i was thinking the same. Were you thinking something like the code below? 

```
const Animation = ({
  children,
  direction,
  duration,
  state,
}) => {
    const defaultStyle = {
      transition: `transform ${duration}ms ease-in-out`,
    };
  
  return (
    <Transition in={state} timeout={duration}>
      {(state) => (
        <div
          style={{
            ...defaultStyle,
            ...directions[direction][state],
          }}
        >
          {children}
        </div>
      )}
    </Transition>
  );
}

export const SlideFromLeftToRight = ({ state, duration, children }) => {
  return (
    <Animation
      children={children}
      direction='leftToRight'
      duration={duration}
      state={state}
    />
  );
};

export const SlideFromRightToLeft = ({ state, duration, children }) => {
  return (
    <Animation
      children={children}
      direction='rightToLeft'
      duration={duration}
      state={state}
    />
  );
};
```

Reply:
Yes, I was thinking of these two parts: 'SlideFromLeftToRight' and 'SlightFromRightToLeft'.